app/src/main/java/com/beemdevelopment/aegis/services/LaunchScannerTileService.java

```diff
@@ -0,0 +1,61 @@
+package com.beemdevelopment.aegis.services;
+
+import android.Manifest;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Build;
+import android.service.quicksettings.Tile;
+import android.service.quicksettings.TileService;
+
+import androidx.annotation.RequiresApi;
+
+import com.beemdevelopment.aegis.R;
+import com.beemdevelopment.aegis.ui.MainActivity;
+
+@RequiresApi(api = Build.VERSION_CODES.N)
+public class LaunchScannerTileService extends TileService {
+    private static final int CODE_PERM_CAMERA = 0;
+
+    @Override
+    public void onTileAdded() {
+        super.onTileAdded();
+    }
+
+    @Override
+    public void onStartListening() {
+        super.onStartListening();
+        Tile tile = getQsTile();
+        if(checkCallingPermission(Manifest.permission.CAMERA) != PackageManager.PERMISSION_GRANTED) {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+                tile.setSubtitle(getResources().getString(R.string.tile_no_camera_permission));
+            }
+
+            tile.setState(Tile.STATE_UNAVAILABLE);
```

Comment:
This is a nice idea, but perhaps it's more trouble than it's worth. This makes the tile unclickable. It also doesn't update if I grant the camera permission, even after I forcefully kill the app.